plugins/generator-1.20.1/datapack-1.20.1/features/feature_spring.json.ftl

```diff
@@ -0,0 +1,19 @@
+{
+  "state": {
+    <#if field$state?starts_with("CUSTOM:")>
+      "Name": "${modid}:${field$state?ends_with(":Flowing")?then("flowing_","")}${generator.getRegistryNameFromFullName(field$state)}"
+    <#else>
+      "Name": "${generator.map(field$state, "fluids", 1)}"
+    </#if>
+  },
+  "valid_blocks": ${input$valid_blocks}
+  <#if field$requires_block_below != "TRUE">,

```

Comment:
So far we check the lower-cased values of boolean fields in most (if not all) places.

Reply:
There's a similar check in `projectiles_arrow.java.ftl`, and converting to lowercase seems an unnecessary step (which we could probably remove from other templates?)

Reply:
If we are sure these bool values are always in uppercase, then I think it is fine to leave it as is (and ideally fix existing templates to not lowercase in another PR)

Reply:
According to the [Blockly Guide](https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/checkbox#xml), the value of a checkbox field is always either `TRUE` or `FALSE`

Reply:
Then this is ok, and PR fixing this for other blocks is also welcome :)